src/spring/azext_spring/_utils.py

```diff
@@ -207,7 +208,7 @@ def get_azure_files_info(file_sas_url):
 
 
 def _get_azure_storage_client_info(account_type, sas_url):
-    regex = compile("http(s)?://(?P<account_name>.*?)\.{0}\.(?P<endpoint_suffix>.*?)/(?P<container_name>.*?)/(?P<relative_path>.*?)\?(?P<sas_token>.*)".format(account_type))
+    regex = compile(f'http(s)?://(?P<account_name>.*?){re.escape(".")}{account_type}{re.escape(".")}(?P<endpoint_suffix>.*?)/(?P<container_name>.*?)/(?P<relative_path>.*?){re.escape("?")}(?P<sas_token>.*)')
```

Comment:
This regular expression looks very complex. Could you add examples to it in the comments to increase its readability?

Reply:
(1) Add some comments (2) Add one more test in newly added file `src/spring/azext_spring/tests/latest/test_utils.py`